src/ImportDefinitionsBundle/Importer/Importer.php

```diff
@@ -153,12 +153,12 @@ public function doImport(DefinitionInterface $definition, $params)
         /** @var ImportDataSetInterface|array $data */
         $data = $this->getData($definition, $params);
 
-        if (\count($data) > 0) {
+        if ((is_array($data) || $data instanceof \Countable) && \count($data) > 0) {
             $this->eventDispatcher->dispatch($definition, 'import_definition.total', \count($data), $params);
-
-            list($objectIds, $exceptions) = $this->runImport($definition, $params, $filter, $data);
         }
 
+        list($objectIds, $exceptions) = $this->runImport($definition, $params, $filter, $data);
```

Comment:
We always run the import, even if we cannot count it here.

Reply:
not sure about this, since we dropped the generator, we know the count right?

Reply:
Yes, but this is still the correct thing to do since not all data sources will be countable.

Reply:
but currently we assume they are. We should find another way to display a progress instead, not sure, but I think that symfony progress doesn't support infinite scroll right?

Reply:
I asked and apparently they do, you just not set the max.

Reply:
We can move that (with this bugfix) into a separate PR where it's fully fixed, supporting uncountable providers.

Reply:
Yes. Let's do that. Your PR is fine then.